### PR TITLE
gnome3.geary: 0.12.4 -> 0.13.1

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Ends the litany of patches and other horrible things :smile:

[Link to News](https://gitlab.gnome.org/GNOME/geary/blob/8afe3c6f24abc580c05b047eba8198ba0ae987e9/NEWS)
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
I think the test failure on aarch might have been reported upstream.

File: pkgs/desktops/gnome-3/misc/geary/default.nix [hunk 1]
```diff
@@ -1,62 +1,56 @@
...
-      sha256 = "0d5hc4h9c1hnn2sk18nkpmzdvwm3h746n2zj8n22ax9rj6lxl38l";
-    })
-    # Fix build with vala 0.40.12
-    # See: https://gitlab.gnome.org/GNOME/vala/blob/0.40.12/NEWS#L22
-    (fetchpatch {
-      url = "https://gitlab.gnome.org/GNOME/geary/commit/088cb2c0aa35ad4b54ea5a0a2edaf0ff96c64b45.patch";
-      sha256 = "0cnjmbd3snm8ggmprqa32f7i3w86gs3ylab9p5ffj921dcpvvlb2";
-    })
+  nativeBuildInputs = [
+    desktop-file-utils gettext gobject-introspection itstool
```
Comment (jtojnar) [inline pkgs/desktops/gnome-3/misc/geary/default.nix]: Do we still need `gobject-introspection` for vala apps with https://github.com/NixOS/nixpkgs/pull/54324?

Comment (worldofpeace) [reply pkgs/desktops/gnome-3/misc/geary/default.nix]: This is a rare case https://gitlab.gnome.org/GNOME/geary/blob/master/meson.build#L63

We probably should do a treewide cleanup for that.

**Comment by jtojnar** [comment]
I am getting `TLS support is not available`, we need `glib-networking`. Also, I got rid of few unnecessary dependencies:

```patches
diff --git a/pkgs/desktops/gnome-3/misc/geary/default.nix b/pkgs/desktops/gnome-3/misc/geary/default.nix
index d9ca60ddcde..074a9c90f67 100644
--- a/pkgs/desktops/gnome-3/misc/geary/default.nix
+++ b/pkgs/desktops/gnome-3/misc/geary/default.nix
@@ -1,8 +1,8 @@
-{ stdenv, fetchurl, pkgconfig, gtk3, vala, enchant, wrapGAppsHook, gdk_pixbuf, meson, ninja
+{ stdenv, fetchurl, fetchpatch, pkgconfig, gtk3, vala, enchant2, wrapGAppsHook, meson, ninja
 , desktop-file-utils, gnome-online-accounts, gsettings-desktop-schemas, adwaita-icon-theme
-, libnotify, libcanberra-gtk3, libsecret, gmime, isocodes, libxml2, gettext, gobject-introspection
-, libpthreadstubs, sqlite, gcr, json-glib, itstool, libgee, gnome3, librsvg, webkitgtk, python3
-, xvfb_run, dbus, shared-mime-info, libunwind }:
+, libnotify, libcanberra-gtk3, libsecret, gmime, isocodes, libxml2, gettext
+, sqlite, gcr, json-glib, itstool, libgee, gnome3, webkitgtk, python3
+, xvfb_run, dbus, shared-mime-info, libunwind, glib-networking }:
 
 stdenv.mkDerivation rec {
   pname = "geary";
@@ -13,15 +13,24 @@
     sha256 = "0h9pf2mdskq7mylib1m9hw86nwfmdzyngjl7ywangqipm1k5svjx";
   };
 
+  patches = [
+    # gobject-introspection is not needed
+    # https://gitlab.gnome.org/GNOME/geary/merge_requests/138
+    (fetchpatch {
+      url = https://gitlab.gnome.org/GNOME/geary/commit/d2f1b1076aa942d140e83fdf03b66621c11229f5.patch;
+      sha256 = "1dsj4ybnibpi572w9hafm0w90jbjv7wzdl6j8d4c2qg5h7knlvfk";
+    })
+  ];
+
   nativeBuildInputs = [
-    desktop-file-utils gettext gobject-introspection itstool
+    desktop-file-utils gettext itstool
     libxml2 meson ninja pkgconfig vala wrapGAppsHook python3
   ];
 
   buildInputs = [
-    adwaita-icon-theme enchant gcr gdk_pixbuf gmime gnome-online-accounts
+    adwaita-icon-theme enchant2 gcr gmime gnome-online-accounts
     gsettings-desktop-schemas gtk3 isocodes json-glib libcanberra-gtk3
-    libgee libnotify libpthreadstubs librsvg libsecret sqlite webkitgtk
+    libgee libnotify libsecret sqlite webkitgtk glib-networking
     libunwind
   ];
 
```

File: pkgs/desktops/gnome-3/misc/geary/default.nix (hunk 1 above)
Comment (jtojnar) [reply pkgs/desktops/gnome-3/misc/geary/default.nix]: Turns out it is actually just a relic: https://gitlab.gnome.org/GNOME/geary/merge_requests/138

**Comment by worldofpeace** [comment]
Thought `enchant` defaulted to  2.

Didn't look to see if they actually used `gobject-introspection` anymore.
Thanks for the patch.

**Comment by jtojnar** [comment]
> Thought `enchant` defaulted to 2.

It does not; not yet. We probably want to go the Arch way and patch the old apps (the API breakage is minor) but I did not have time to do it yet.
 
> Didn't look to see if they actually used `gobject-introspection` anymore.
> Thanks for the patch.

Yeah, I was wondering what would pure Vala app need `g-i` for. Turns out they used to generate GIR and then Vapi from WebkitGtk headers.

**Comment by worldofpeace** [comment]
> Thought `enchant` defaulted to 2.

> It does not; not yet. We probably want to go the Arch way and patch the old apps (the API breakage is minor) but I did not have time to do it yet.

I'll try to open an issue here, should be sprintable, especially since Arch already has patches.

> Yeah, I was wondering what would pure Vala app need `g-i` for. Turns out they used to generate GIR and then Vapi from WebkitGtk headers.

Right, this app has gone through a lot :smile:

**Comment by worldofpeace** [comment]
Looks like there was a concern for libunwind
- https://gitlab.gnome.org/GNOME/geary/issues/238
- https://gitlab.gnome.org/GNOME/geary/merge_requests/109

Not sure if we'd need to care about that here.

**Comment by jtojnar** [comment]
If it builds on our platforms, I think we are golden.

**Review by jtojnar — APPROVED** [review]

**Pull request merged**